builder/common_builder/common_builder.go

```diff
@@ -113,13 +122,20 @@ func (b *CommonBuilder) checksToRunExadata() {
 
 func (b *CommonBuilder) setOrResetFetcherUser(user string) {
```

Comment:
The cyclomatic complexity of this function is a little bit high, what about something like this:
```
func (b *CommonBuilder) setOrResetFetcherUser(user string) {
	if strings.TrimSpace(user) == "" && runtime.GOOS != "linux" {
		return
	}

	if runtime.GOOS != "linux" {
		b.log.Errorf("Can't set user [%s] for fetcher because it is not supported")
		return
	}

	if strings.TrimSpace(user) == "" {
		if err := b.fetcher.SetUserAsCurrent(); err != nil {
			b.log.Panicf("Can't set current user for fetchers, err: [%v]", user, err)
		}
	} else {
		if err := b.fetcher.SetUser(user); err != nil {
			b.log.Panicf("Can't set user [%s] for fetchers, err: [%v]", user, err)
		}
	}
}
```
?
I also added guards and elevated log for GOOS != linux in error instead of warn...

Reply:
Ok.

Reply:
I think the function is a little more complex to read

Reply:
Is this a little easier?
```go
func (b *CommonBuilder) setOrResetFetcherUser(user string) {
	if runtime.GOOS != "linux" {
		if strings.TrimSpace(user) != "" {
			b.log.Errorf("Can't set user [%s] for fetcher because it is not supported")
		}

		return
	}

	if strings.TrimSpace(user) == "" {
		if err := b.fetcher.SetUserAsCurrent(); err != nil {
			b.log.Panicf("Can't set current user for fetchers, err: [%v]", user, err)
		}
	} else {
		if err := b.fetcher.SetUser(user); err != nil {
			b.log.Panicf("Can't set user [%s] for fetchers, err: [%v]", user, err)
		}
	}
}
```

Reply:
Yes.